src/visualization/types/SimpleTable/PagedTable.tsx

```diff
@@ -278,23 +278,21 @@ const PagedTable: FC<Props> = ({result, properties}) => {
 
   useEffect(() => {
     setSize(size)
-  }, [size])
+  }, [size]) // eslint-disable-line react-hooks/exhaustive-deps
 
   useEffect(() => {
-    if (size > maxSize) {
-      setMaxSize(size)
-    }
-  }, [size])
+    setMaxSize(measurePage(result, 0, height))
+  }, [height, result]) // eslint-disable-line react-hooks/exhaustive-deps
 
   useEffect(() => {
     setPage(1)
-  }, [result])
+  }, [result]) // eslint-disable-line react-hooks/exhaustive-deps
 
   useEffect(() => {
-    if (size) {
-      setTotalPages(Math.ceil((result?.table?.length ?? 0) / size))
+    if (maxSize) {
+      setTotalPages(Math.ceil((result?.table?.length ?? 0) / maxSize))
     }
-  }, [height, result])
+  }, [maxSize]) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Disabling lint ruling here because the warning ask for the setters to be included in the dependencies. For `result?.table?.length`, the entire visualization re-renders when result changes, and given a `result` and it's `table`, the length will remain static. It will only change when result changes, which again, re-renders the entire visualization.

![image](https://user-images.githubusercontent.com/39343323/128386606-6543ab7f-e5bb-4666-aa21-1ad7c94f68f6.png)